src/main/java/com/algorithmia/algorithmHandler/Request.java

```diff
@@ -0,0 +1,23 @@
+package com.algorithmia.algorithmHandler;
+
+class Request<T>{
+    public String content_type;
+    public T data;
+
+    private T CheckCompatability(Object req) throws Exception{
+        T type = null;
+        try{
+            T algoInput = (T)req;
+            return algoInput;
+        } catch (Exception e) {
+            throw new Exception("request " + req.getClass() + " is not valid for apply of type " + type.getClass());
+        }
+    }
+
+    Request(String content_type, Object data) throws Exception{
```

Comment:
Since you've already declared this as a generic class parameterized by `T`, you should be using `T` as the type for `data`, not `Object`

Reply:
It comes in as a blob that we need to massage into the right type (deserialization of binary, encoding into string types, etc.)